Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 929c51c7cf000f094c0259c112b3d4f37e158628
Number of Error(s): 5
/mnt/disk1/home/vamosax/smoketest/PR-12100/HPCC-Platform/ecl/hqlcpp/hqlres.cpp:225:17: error: \xe2\x80\x98cppInstance\xe2\x80\x99 was not declared in this scope

[Build error:](https://gist.github.com/e43267e6d2894a4e273e26be77743dd5)
Build log [Build log](https://gist.github.com/HPCCSmoketest/e43267e6d2894a4e273e26be77743dd5#file-result-19-01-25-11-16-09-log)

--- a/ecl/eclcc/eclcc.cpp
+++ b/ecl/eclcc/eclcc.cpp
@@ -808,13 +808,16 @@ void EclCC::instantECL(EclCompileInstance & instance, IWorkUnit *wu, const char 
                 if (!optShared)
                     wu->setDebugValueInt("standAloneExe", 1, true);
                 EclGenerateTarget target = optWorkUnit ? EclGenerateNone : (optNoCompile ? EclGenerateCpp : optShared ? EclGenerateDll : EclGenerateExe);
-                gatherResourceManifestFilenames(instance, resourceManifestFiles);
-                ForEachItemIn(i, resourceManifestFiles)
-                    generator->addManifest(resourceManifestFiles.item(i));
                 if (instance.srcArchive)
                 {
-                    generator->addManifestFromArchive(instance.srcArchive);
+                    generator->addManifestsFromArchive(instance.srcArchive);
                     instance.srcArchive.clear();
+                }
+                else
+                {
+                    gatherResourceManifestFilenames(instance, resourceManifestFiles);
+                    ForEachItemIn(i, resourceManifestFiles)
+                        generator->addManifest(resourceManifestFiles.item(i));
                 }
                 generator->setSaveGeneratedFiles(optSaveCpp);
 

--- a/ecl/hqlcpp/hqlcpp.hpp
+++ b/ecl/hqlcpp/hqlcpp.hpp
@@ -113,7 +113,7 @@ public:
     virtual void addResource(const char * type, unsigned len, const void * data, IPropertyTree *manifestEntry, unsigned id=(unsigned)-1) = 0;
     virtual void addCompressResource(const char * type, unsigned len, const void * data, IPropertyTree *manifestEntry, unsigned id=(unsigned)-1) = 0;
     virtual void addManifest(const char *filename) = 0;
-    virtual void addManifestFromArchive(IPropertyTree *archive) = 0;
+    virtual void addManifestsFromArchive(IPropertyTree *archive) = 0;
     virtual void addWebServiceInfo(IPropertyTree *wsinfo) = 0;
     virtual void flushHints() = 0;
     virtual void flushResources(const char *filename, ICodegenContextCallback * ctxCallback) = 0;

--- a/ecl/hqlcpp/hqlcpp.ipp
+++ b/ecl/hqlcpp/hqlcpp.ipp
@@ -149,7 +149,7 @@ public:
     virtual void addResource(const char * type, unsigned len, const void * data, IPropertyTree *manifestEntry=NULL, unsigned id=(unsigned)-1);
     virtual void addCompressResource(const char * type, unsigned len, const void * data, IPropertyTree *manifestEntry=NULL, unsigned id=(unsigned)-1);
     virtual void addManifest(const char *filename){resources.addManifest(filename);}
-    virtual void addManifestFromArchive(IPropertyTree *archive){resources.addManifestFromArchive(archive);}
+    virtual void addManifestsFromArchive(IPropertyTree *archive){resources.addManifestsFromArchive(archive);}
     virtual void addWebServiceInfo(IPropertyTree *wsinfo){resources.addWebServiceInfo(wsinfo);}
     virtual void getActivityRange(unsigned cppIndex, unsigned & minActivityId, unsigned & maxActivityId);
     

--- a/ecl/hqlcpp/hqlecl.cpp
+++ b/ecl/hqlcpp/hqlecl.cpp
@@ -92,7 +92,7 @@ public:
     virtual bool generatePackage(const char * packageName);
     virtual void setMaxCompileThreads(unsigned value) { defaultMaxCompileThreads = value; }
     virtual void addManifest(const char *filename) { code->addManifest(filename); }
-    virtual void addManifestFromArchive(IPropertyTree *archive) { code->addManifestFromArchive(archive); }
+    virtual void addManifestsFromArchive(IPropertyTree *archive) { code->addManifestsFromArchive(archive); }
     virtual void addWebServiceInfo(IPropertyTree *wsinfo){ code->addWebServiceInfo(wsinfo); }
 
     virtual double getECLcomplexity(IHqlExpression * exprs);

--- a/ecl/hqlcpp/hqlecl.hpp
+++ b/ecl/hqlcpp/hqlecl.hpp
@@ -46,7 +46,7 @@ public:
     virtual bool processQuery(OwnedHqlExpr & parsedQuery, EclGenerateTarget generateTarget) = 0;
     virtual void setMaxCompileThreads(unsigned max) = 0;
     virtual void addManifest(const char *filename) = 0;
-    virtual void addManifestFromArchive(IPropertyTree *archive) = 0;
+    virtual void addManifestsFromArchive(IPropertyTree *archive) = 0;
     virtual void addWebServiceInfo(IPropertyTree *wsinfo) = 0;
     virtual void setSaveGeneratedFiles(bool value) = 0;
 };

--- a/ecl/hqlcpp/hqlres.cpp
+++ b/ecl/hqlcpp/hqlres.cpp
@@ -92,6 +92,8 @@ static void loadResource(const char *filepath, MemoryBuffer &content)
 {
     Owned <IFile> f = createIFile(filepath);
     Owned <IFileIO> fio = f->open(IFOread);
+    if (!fio)
+        throw makeStringExceptionV(0, "Failed to open resource file %s", filepath);
     read(fio, 0, (size32_t) f->size(), content);
 }
 
@@ -246,7 +248,7 @@ void ResourceManager::addManifestInclude(IPropertyTree &include, const char *dir
     addManifestFile(includePath.str());
 }
 
-void ResourceManager::addManifestFromArchive(IPropertyTree *archive)
+void ResourceManager::addManifestsFromArchive(IPropertyTree *archive)
 {
     if (!archive)
         return;

--- a/ecl/hqlcpp/hqlres.hpp
+++ b/ecl/hqlcpp/hqlres.hpp
@@ -32,7 +32,7 @@ public:
     void addNamed(const char * type, unsigned len, const void *data, IPropertyTree *entry=NULL, unsigned id=(unsigned)-1, bool addToManifest=true, bool compressed=false);
     bool addCompress(const char * type, unsigned len, const void *data, IPropertyTree *entry=NULL, unsigned id=(unsigned)-1, bool addToManifest=true);
     void addManifest(const char *filename);
-    void addManifestFromArchive(IPropertyTree *archive);
+    void addManifestsFromArchive(IPropertyTree *archive);
     void addWebServiceInfo(IPropertyTree *wsinfo);
     IPropertyTree *ensureManifestInfo(){if (!manifest) manifest.setown(createPTree("Manifest")); return manifest;}
     bool getDuplicateResourceId(const char *srctype, const char *respath, const char *filename, int &id);

